### PR TITLE
Feat/add robots.txt file

### PR DESCRIPTION
**Pull request by vdusart**
## Description
This PR adds the generation of the `robots.txt` file inside the `gatsby-config.ts` file.

After the build the generated file looks like that:
```
User-agent: *
Allow: /
Sitemap: https://ethereum.org/sitemap/sitemap-index.xml
Host: https://ethereum.org
```

I didn't know what policy do you want to have so I kept the default one (`policy: [{ userAgent: "*", allow: "/" }]`), tell me if you want to change that.

## Related Issue
Add feature requested in #7464

### CODE REVIEW

File: gatsby-config.ts [hunk 1]
```diff
@@ -104,6 +104,15 @@ const config: GatsbyConfig = {
         },
       },
     },
+    // robots.txt creation
+    {
+      resolve: "gatsby-plugin-robots-txt",
+      options: {
+        host: "https://ethereum.org/",
+        sitemap: "https://ethereum.org/sitemap/sitemap-index.xml",
```
Comment (pettinarip) [inline gatsby-config.ts:112]: There is a local variable called `siteUrl` at the beginning of this file. Lets try to use it here just as a good practice. What do you think?

**Review by pettinarip — CHANGES_REQUESTED** [review]
Nice @vdusart, thanks for doing this. Left a comment.

**Review by wackerow — CHANGES_REQUESTED** [review]
@vdusart Nice! Thanks for doing this... I agree with Pablo, if we could reuse the `siteUrl` variable both for the `host` and for the beginning of the `sitemap` string.

Also, one small favor to ask, would you mind updating the comment in the section _above_ your `gatsby-config` changes? The comment says:
```
// Sitemap generator (ethereum.org/sitemap.xml)
```
Would you mind updating this to:
```
// Sitemap generator (ethereum.org/sitemap/sitemap-index.xml)
```
🙏 Thanks again!

File: gatsby-config.ts (hunk 1 above)
Comment (vdusart) [reply gatsby-config.ts:112]: Of course I'm ok, I missed that !

**Comment by vdusart** [comment]
> @vdusart Nice! Thanks for doing this... I agree with Pablo, if we could reuse the `siteUrl` variable both for the `host` and for the beginning of the `sitemap` string.
> 
> Also, one small favor to ask, would you mind updating the comment in the section _above_ your `gatsby-config` changes? The comment says:
> 
> ```
> // Sitemap generator (ethereum.org/sitemap.xml)
> ```
> 
> Would you mind updating this to:
> 
> ```
> // Sitemap generator (ethereum.org/sitemap/sitemap-index.xml)
> ```
> 
> pray Thanks again!



> @vdusart Nice! Thanks for doing this... I agree with Pablo, if we could reuse the `siteUrl` variable both for the `host` and for the beginning of the `sitemap` string.
> 
> Also, one small favor to ask, would you mind updating the comment in the section _above_ your `gatsby-config` changes? The comment says:
> 
> ```
> // Sitemap generator (ethereum.org/sitemap.xml)
> ```
> 
> Would you mind updating this to:
> 
> ```
> // Sitemap generator (ethereum.org/sitemap/sitemap-index.xml)
> ```
> 
> pray Thanks again!

Yes no problem, I'm on it :smile:

**Review by pettinarip — APPROVED** [review]

**Review by wackerow — APPROVED** [review]
Thanks @vdusart! 💪

**Pull request merged**